### PR TITLE
[Console] Added fluent PHP-DSL for configuring commands

### PR DESCRIPTION
**Pull request by atailouloute**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       |  https://github.com/orgs/symfony/projects/1#card-31472643
| License       | MIT
| Doc PR        | TODO

Added fluent PHP-DSL for configuring commands like DI and routes

An example:

```php
<?php

use Symfony\Component\Console\Application;
use Symfony\Component\Console\Configurator\ConsoleConfigurator;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;

require_once __DIR__ . '/vendor/autoload.php';

$application = new Application();
$configurator = new ConsoleConfigurator($application);

$configurator
    ->add('great')
        ->processTitle('title')
        ->description('This is a description for "great" command')
        ->aliases('say_hello')
        ->usage('ahmed')
        // adds another usage example
        ->usage('ahmed nicolas')

        ->argument('names')
            ->optional()
            ->default([])
            ->array()

        ->option('upper', 'u')
            ->description('uppercase names')

        ->code(function (InputInterface $input, OutputInterface $output) {
            $names = implode(', ', $input->getArgument('names'));

            $output->writeln('Hello ' . ($input->getOption('upper') ? strtoupper($names) : $names));
        })
;

$configurator
    ->name('Console')
    ->version('1.1.1')
    ->defaultCommand('help')
;


$application->run();
```

### CODE REVIEW

**Comment by chalasr** [comment]
What is the target project type for this configurator approach? 
Is it about using `symfony/console` solely to build console-only applications?
Or "full-stack" applications (framework) providing some console commands?

Since 3.4 we push toward having a separation between the command routing and its definition+behavior, mainly because of commands as services and lazy loading, because console commands have dependencies and should deserve clean design thanks to dependency injection.

So, basically, what about dependency injection and lazy loading when using such a configurator?

**Comment by Nyholm** [comment]
Friendly ping

**Comment by fabpot** [comment]
I don't see the usefulness of this new feature. Let's close.
